rpcs3/Input/pad_thread.cpp

```diff
@@ -311,11 +311,10 @@ void pad_thread::operator()()
 		{
 			if (thread)
 			{
-				auto& enumeration_thread = *thread;
-				enumeration_thread = thread_state::aborting;
-				enumeration_thread();
+				thread->reset();
```

Comment:
```suggestion
				thread.reset();
```

Reply:
Hmmm. Wouldn't the threads.clear suffice anyway?

Reply:
here you reset them in order explicitly, std::vector on the other hand calls destructors in unspecified order. (matters for debugging purposes)